src/main/java/com/fortanix/sdkms/performance/sampler/AbstractSignatureSampler.java

```diff
@@ -76,8 +73,8 @@ public void setupTest(JavaSamplerContext context) {
             LOGGER.log(Level.INFO, "failure in creating key : " + e.getMessage(), e);
             throw new ProviderException(e.getMessage());
         }
-        this.signRequest = new SignRequest().hashAlg(HASH_ALGORITHM).hash(this.hash);
-        this.signRequestEx = new SignRequestEx().hashAlg(HASH_ALGORITHM).hash(this.hash).key(new SobjectDescriptor().kid(this.keyId));
+        this.signRequest = new SignRequest().hashAlg(digestAlgorithm).hash(this.hash);
+        this.signRequestEx = new SignRequestEx().hashAlg(digestAlgorithm).data(input.getBytes()).key(new SobjectDescriptor().kid(this.keyId));
```

Comment:
Related to above comment. Better to send hash to Sign API instead of input data.

Reply:
fixed